packages/mapper/src/mapper.ts

```diff
@@ -78,12 +78,26 @@ export default class Mapper {
     this.messageBus.emit('mongosh:api-call', event);
   }
 
-  use(db): any {
-    if (!(db in this.databases)) {
-      this.databases[db] = new Database(this, db);
+  private _getDatabase(name: string): Database {
+    if (typeof name !== 'string') {
+      throw new MongoshInvalidInputError(
+        `Database name must be a string. Received ${typeof name}.`);
+    }
+
+    if (!name.trim()) {
+      throw new MongoshInvalidInputError('Database name cannot be empty.');
+    }
+
+    if (!(name in this.databases)) {
```

Comment:
Should we do something about case sensitivity here? https://docs.mongodb.com/manual/reference/limits/#Database-Name-Case-Sensitivity

Reply:
hmmm as I was answering this, I got a bit confused by this method. The name suggests a `read` from the driver to me, but I think you're just adding a new database to `this.databases`, which is a `write`. If this is indeed a way to save a database in our local state, I think we should handle it. Really good catch finding this, I totally didn't know case sensitivity was a thing.

My original answer was:
I think this is something we should be doing for a `write` if the driver _does not_ handle it for us (I feel like it would). If it's just a `read` method, than we should have handled it before we wrote it.

Reply:
is `_getDatabase` a confusing name maybe? What about `_getAndCacheDatabase`?